benchmarks/bench_algos.py

```diff
@@ -0,0 +1,207 @@
+import pytest
+
+import cugraph
+from cugraph.tests import utils
+
+import pytest_benchmark
+# FIXME: Remove this when rapids_pytest_benchmark.gpubenchmark is available
+# everywhere
```

Comment:
What would be a good path to optionally enable managed memory?

Reply:
`pytest-benchmark` lets benchmark authors add their own command-line options, which I'm not doing here.  I can add one for managed memory and have a fixture look for it accordingly though. Do you think that's a good way to go?

Reply:
I added new markers for fine-grained control of how RMM reinit happens, and a new command-line option (`--no-rmm-reinit`) to disable it completely.